hr_timesheet_sheet_no_negative/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models, _
+
+
+class SheetLine(models.TransientModel):
+    _inherit = 'hr_timesheet.sheet.line'
+
+    def _negative_unit_amount(self):
+        if not self.sheet_id.company_id:
+            user_company = self.env.user.company_id
+            company_id = self._context.get('company_id') or user_company.id
+            company = self.env['res.company'].browse(company_id)
+        else:
+            company = self.sheet_id.company_id
+        allow_negative = company.timesheet_negative_unit_amount
+        return allow_negative
+
+    @api.onchange('unit_amount')
+    def onchange_unit_amount(self):
+        self.ensure_one()
+        sheet = self._get_sheet()
+        if not sheet:
+            return {'warning': {
+                'title': _("Warning"),
+                'message': _("Save the Timesheet Sheet first.")
+            }}
+        elif not self.sheet_id:
+            self.sheet_id = sheet
```

Comment:
I see 8 lines of code duplicated from the main module. There's no better solution? :thinking:

Reply:
Any suggestion how to do, please?

Reply:
removed...